source/evp_hash.c

```diff
@@ -0,0 +1,344 @@
+
+    if (!SHA224_Init(&sha256_ctx))
+        return 0;
+
+    for (i = 0; i < update_times; i++)
+        if (!SHA224_Update(&sha256_ctx, data, sizeof(data)))
+            return 0;
+
+    return SHA224_Final(md, &sha256_ctx);
+}
+
+int hash_sha256_deprecated()
+{
+    int i;
+    SHA256_CTX sha256_ctx;
+    unsigned char md[EVP_MAX_MD_SIZE];
+
+    if (!SHA256_Init(&sha256_ctx))
+        return 0;
+
+    for (i = 0; i < update_times; i++)
+        if (!SHA256_Update(&sha256_ctx, data, sizeof(data)))
+            return 0;
+
+    return SHA256_Final(md, &sha256_ctx);
+}
+
+int hash_sha384_deprecated()
+{
+    int i;
+    SHA512_CTX sha512_ctx;
+    unsigned char md[EVP_MAX_MD_SIZE];
+
+    if (!SHA384_Init(&sha512_ctx))
+        return 0;
+
+    for (i = 0; i < update_times; i++)
+        if (!SHA384_Update(&sha512_ctx, data, sizeof(data)))
+            return 0;
+
+    return SHA384_Final(md, &sha512_ctx);
+}
+
+int hash_sha512_deprecated()
+{
+    int i;
+    SHA512_CTX sha512_ctx;
+    unsigned char md[EVP_MAX_MD_SIZE];
+
+    if (!SHA512_Init(&sha512_ctx))
+        return 0;
+
+    for (i = 0; i < update_times; i++)
+        if (!SHA512_Update(&sha512_ctx, data, sizeof(data)))
+            return 0;
+
+    return SHA512_Final(md, &sha512_ctx);
+}
+
+
+int hash_evp(EVP_MD_CTX *mctx, const EVP_MD *evp_md)
+{
+    int i;
+    unsigned char md[EVP_MAX_MD_SIZE];
+
+    if (!EVP_DigestInit(mctx, evp_md))
+        return 0;
+
+    for (i = 0; i < update_times; i++)
+        if (!EVP_DigestUpdate(mctx, data, sizeof(data)))
+            return 0;
+
+    return EVP_DigestFinal(mctx, md, NULL);
+}
+
+void do_hash_deprecated(size_t num)
+{
+    OSSL_TIME time;
+
+    do {
+        if (!hash_func_deprecated()) {
+            err = 1;
+            return;
+        }
+
+        counts[num]++;
+        time = ossl_time_now();
+    } while (time.t < max_time.t);
+}
+
+void do_hash_evp(size_t num)
+{
+    OSSL_TIME time;
+    EVP_MD_CTX *mctx = EVP_MD_CTX_new();
+
+    if (mctx == NULL) {
+        err = 1;
+        return;
+    }
+
+    do {
+        if (!hash_evp(mctx, evp_md)) {
+            err = 1;
+            goto err;
+        }
+ 
+        counts[num]++;
+        time = ossl_time_now();
+    } while (time.t < max_time.t);
+
+err:
+    EVP_MD_CTX_free(mctx);
+}
+
+void print_help()
+{
+    printf("Usage: evp_hash [-h] [-x] [-t] [-u update-times] [-a algorithm] thread-count\n");
+    printf("-h - print this help output\n");
+    printf("-x - use deprecated API instead of EVP API\n");
+    printf("-t - terse output\n");
+    printf("-u update-times - times to update digest. 1 for one-shot (default: 1)\n");
+    printf("-a algorithm - One of: [SHA1, SHA224, SHA256, SHA384, SHA512] (default: SHA1)\n");
+    printf("thread-count - number of threads\n");
+}
+
+int main(int argc, char *argv[])
+{
+    OSSL_TIME duration;
+    size_t total_count = 0;
+    double av;
+    int terse = 0, deprecated_api = 0, hash_algorithm = SHA1_ALG;
```

Comment:
There is no reason to set the value here (even if it sets 0 basically);  it's confusing `hash_algorithm` having a default value, when it gets overridden always anyway